pkg/virt-handler/isolation/isolation.go

```diff
@@ -152,48 +168,91 @@ func (s *socketBasedIsolationDetector) AdjustResources(vm *v1.VirtualMachineInst
 		return nil
 	}
 
-	// bump memlock ulimit for libvirtd
-	res, err := s.Detect(vm)
+	isolationResult, err := s.Detect(vm)
+	if err != nil {
+		return err
+	}
+
+	// make the best estimate for memory required by libvirt
+	memlockSize, err := getMemlockSize(vm)
 	if err != nil {
 		return err
 	}
-	launcherPid := res.Pid()
 
 	processes, err := ps.Processes()
 	if err != nil {
 		return fmt.Errorf("failed to get all processes: %v", err)
 	}
 
-	for _, process := range processes {
-		// consider all processes that are virt-launcher children
-		if process.PPid() != launcherPid {
-			continue
-		}
-
-		// libvirtd process sets the memory lock limit before fork/exec-ing into qemu
-		if process.Executable() != "libvirtd" {
-			continue
-		}
+	// bump libvirtd and qemu processes locked memory limits
+	libvirtProcesses := filterVirtLaunchersLibvirtProcesses(isolationResult, processes)
+	log.Log.Info("debug: adjust memlock: filtered processes:")
+	for _, process := range libvirtProcesses {
+		log.Log.Infof("debug: pid %d, ppid %d, exec %s", process.Pid(), process.PPid(), process.Executable())
```

Comment:
please make the debug prints more informative. Something like: "found libvirt process with pid %d...."